### PR TITLE
Add cluster health cmd

### PR DESCRIPTION
**Pull request by cholcombe973**
Also use RadosResult which is less typing

### CODE REVIEW

File: src/cmd.rs [hunk 1]
```diff
@@ -119,6 +119,71 @@ pub struct Mon {
     pub addr: String,
 }
 
+#[derive(Deserialize, Debug)]
+pub enum HealthStatus {
```
Comment (ChrisMacNaughton) [inline src/cmd.rs:123]: maybe more like:

```rust
pub enum Health {
  #[serde...]
  Err,
  #[serde...]
  Warn,
  ..
}
```

File: src/cmd.rs [hunk 2]
```diff
@@ -119,6 +119,71 @@ pub struct Mon {
...
+pub struct ClusterHealth {
+    pub health: Health,
+    pub timechecks: TimeChecks,
+    pub summary: Vec<String>,
+    pub overall_status: HealthStatus,
+    pub detail: Vec<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct Health {
```
Comment (ChrisMacNaughton) [inline src/cmd.rs:142]: Renaming `HealthStatus` to `Health` would require renaming this to something, maybe `HealthServices`

File: src/cmd.rs [hunk 3]
```diff
@@ -119,6 +119,71 @@ pub struct Mon {
...
+    pub health_services: Vec<ServiceHealth>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct TimeChecks {
+    pub epoch: u64,
+    pub round: u64,
+    pub round_status: String,
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: Should round_status be an Enum as well?

File: src/cmd.rs [hunk 4]
```diff
@@ -119,6 +119,71 @@ pub struct Mon {
...
+    pub name: String,
+    pub kb_total: u64,
+    pub kb_used: u64,
+    pub kb_avail: u64,
+    pub avail_percent: u8,
+    pub last_updated: String,
+    pub store_stats: StoreStats,
+    pub health: HealthStatus,
+}
+#[derive(Deserialize, Debug)]
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: Missing newline

File: src/cmd.rs [hunk 5]
```diff
@@ -419,7 +484,49 @@ impl AsRef<str> for PoolOption {
     }
 }
 
-pub fn osd_out(cluster_handle: rados_t, osd_id: u64, simulate: bool) -> Result<(), RadosError> {
+impl fmt::Display for HealthStatus {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &HealthStatus::HealthErr => write!(f, "HEALTH_ERR"),
```
Comment (ChrisMacNaughton) [inline src/cmd.rs]: `write!(f, self.as_ref())` ?

**Review by ChrisMacNaughton — CHANGES_REQUESTED** [review]
Generally OK, some minor nits

File: src/cmd.rs (hunk 1 above)
Comment (cholcombe973) [reply src/cmd.rs:123]: Yeah that looks good

File: src/cmd.rs (hunk 3 above)
Comment (cholcombe973) [reply src/cmd.rs]: Yup.  Done

File: src/cmd.rs (hunk 4 above)
Comment (cholcombe973) [reply src/cmd.rs]: Done

File: src/cmd.rs (hunk 5 above)
Comment (cholcombe973) [reply src/cmd.rs]: Compiler says no dice.  Must be a string literal

Comment (ChrisMacNaughton) [reply src/cmd.rs]: so ... `write!(f, "{}", self.as_ref());` ?

**Review by ChrisMacNaughton — APPROVED** [review]

**Pull request merged**